### DISCUSSION TITLE
SSG: getStaticPaths based on subdomain

### DISCUSSION THREAD

**Discussion by kidrecursive** [vercel/next.js#18142] [Help] [answered]
Hello, I have a nextjs project that loads theming/configuration from a backend for sites based on subdomain, for example: `https://site1.example.com`. I'm hoping to generate these sites statically, however I haven't been able to figure out how to do so. 

I have tried using a custom server that rewrites `https://site1.example.com` to `https://example.com/sites/site1` wherein my next.js app can use getStaticPaths as expected, but this causes problems with routing on the client side after rendering.

Has anyone else had to accomplish the same thing using next.js/getStaticProps?

**Comment by ahmedB7r** [comment, marked as answer]
any updates with this?

**Reply by nyn0x** [reply]
https://github.com/vercel/next.js/discussions/12848#discussioncomment-680225
